File: Week 2 - 12-20-2021/ifElseIfStatementDemo.py
"""
    multiple alternative decision structures are if / else if statements
    
     structure:
    
        if condition:
            statement(s)
        elif condition:
            statement(s)
            
            (can have as many else ifs between an if and an else)
            
        else:
            statement(s)
            
    NO NEED TO PUT A CONDITION AFTER THE WORD ELSE
    (python understands that else is where the if is FALSE)
    
"""
# have the user input a number
a = int(input("Enter a number: "))
# have the user input another number
b = int(input("Enter another number: "))

# use an if / else if to tell the relationship between the numbers
if a > b:
    # first number is greater than the second
    print(f"{a} is greater than {b}")
elif b > a:
    # second number is greater than the first
    print(f"{b} is greater than {a}")
else:
    # both numbers are equal
    print(f"{a} is equal to {b}")
    

File: Week 4 - 01-10-2022/ifElseIfStatementDemo.cpp
#include <iostream>
using namespace std;

int main()
{
	/*
		multiple alternative decision structures have multiple forms -
		this program demonstrates if / else if statements
		
		structure of if / else if statements:
			
				if(condition)
				{
					statement(s); // execute if the expression is true
				}
				else if(condition)
				{
					statement(s); // execute if the expression is true	
				}
				else
				{
					statement(s); // execute when none of the above expressions are true
				}
				
		** you can have as many else ifs between an if and an else as you need **
	*/	
	
	// declare two variables to hold numbers to be input by the user
	int a, b;
	
	// ask the user to enter the numbers
	cout << "Enter a number: ";
	cin >> a;
	cout << "Enter another number: ";
	cin >> b;
	
	// use an if / else if to tell the relationship between the two numbers
	if(a > b)
	{
		// if the first number is greater than the second
		cout << a << " is greater than " << b;
	}
	else if(b > a)
	{
		// if the second number is greater than the first
		cout << b << " is greater than " << a; 
	}
	else
	{
		// otherwise, the numbers must be equal
		cout << a << " is equal to " << b;
	}		
}

